Assignment panics when join before chainstart

Validator client gets a panic when joins before chain start and get assignment, to reproduce:

1.) Start beacon node, say chain start is at `t1`
2.) Start validator client before `t1`
3.) Validator client wait till `t1` to start its duties
4.) Validator client gets a panic right at `t1`

Panic log:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x170c6d9]

goroutine 51 [running]:
github.com/prysmaticlabs/prysm/validator/client.(*validator).RolesAt(0xc000202120, 0x0, 0xc000123650)
	validator/client/validator.go:260 +0x49
github.com/prysmaticlabs/prysm/validator/client.run(0x19cf220, 0xc00006e7c0, 0x19d9220, 0xc000202120)
	validator/client/runner.go:79 +0x608
created by github.com/prysmaticlabs/prysm/validator/client.(*ValidatorService).Start
	validator/client/service.go:109 +0xaa5
```

--- a/validator/client/validator.go
+++ b/validator/client/validator.go
@@ -196,9 +196,6 @@ func (v *validator) UpdateAssignments(ctx context.Context, slot uint64) error {
 	}
 	ctx, span := trace.StartSpan(ctx, "validator.UpdateAssignments")
 	defer span.End()
-	if slot == 0 {
-		return nil
-	}
 
 	req := &pb.AssignmentRequest{
 		EpochStart: slot / params.BeaconConfig().SlotsPerEpoch,

